cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpPlatformResources.java

```diff
@@ -184,18 +192,25 @@ public CloudNetworks networks(CloudCredential cloudCredential, Region region, Ma
         return new CloudNetworks(result);
     }
 
-    public SubnetworkList getSubnetworkList(Region region, Compute compute, String projectId, String subnetId, String sharedProjectId) throws IOException {
+    public SubnetworkList getSubnetworkList(Region region, Compute compute, String projectId, List<String> subnetIds,
+        String sharedProjectId) throws IOException {
         SubnetworkList subnetworkList;
-        if (StringUtils.isEmpty(subnetId)) {
+        if (subnetIds != null && subnetIds.isEmpty()) {
             subnetworkList = compute.subnetworks()
                     .list(projectId, region.value()).execute();
         } else {
             if (!Strings.isNullOrEmpty(sharedProjectId)) {
-                subnetworkList = new SubnetworkList()
-                        .setItems(Collections.singletonList(compute.subnetworks().get(sharedProjectId, region.value(), subnetId).execute()));
+                subnetworkList = new SubnetworkList().setItems(new ArrayList<>());
+                for (String subnetId : subnetIds) {
+                    Subnetwork subnetwork = compute.subnetworks().get(sharedProjectId, region.value(), subnetId).execute();
+                    subnetworkList.getItems().add(subnetwork);
+                }
             } else {
-                subnetworkList = new SubnetworkList()
-                        .setItems(Collections.singletonList(compute.subnetworks().get(projectId, region.value(), subnetId).execute()));
+                subnetworkList = new SubnetworkList().setItems(new ArrayList<>());
+                for (String subnetId : subnetIds) {
+                    Subnetwork subnetwork = compute.subnetworks().get(projectId, region.value(), subnetId).execute();
+                    subnetworkList.getItems().add(subnetwork);
+                }
             }
         }
         return subnetworkList;
```

Comment:
this seems a bit complicated and could be simplified. Like instead of query every subnet one by one we could still use the list then filter the result.
```
    private SubnetworkList getSubnetworkList(Region region, Compute compute, String projectId, List<String> subnetIds,
        String sharedProjectId) throws IOException {
        String projectIdToQuery = StringUtils.isNotBlank(sharedProjectId) ? sharedProjectId : projectId;
        SubnetworkList subnetworkList = compute.subnetworks().list(projectIdToQuery, region.value()).execute();
        if (subnetIds.isEmpty()) {
            return subnetworkList;
        } else {
            List<Subnetwork> filteredSubnets = subnetworkList.getItems().stream()
                    .filter(subnetwork -> subnetIds.contains(subnetwork.getName()))
                    .collect(Collectors.toList());
            return new SubnetworkList().setItems(filteredSubnets);
        }
    }

Reply:
unfortunately it is not the same here? They are using the same object but in case of list most of the field is null

Reply:
yeah, right, instead of creating a new, the old one's items should be overridden, that should do the trick. but the old implementation did the same, creating a new `SubnetworkList` object and adding subnetoworks to it's items.

Reply:
We just discussed on zoom